### ISSUE TITLE
`Errno::ECONNRESET: Connection reset by peer` followed by `Errno::EPIPE: Broken pipe`

### ISSUE THREAD

**Issue by adamwaite** [thoughtbot/capybara-webkit#609]
Hello.

Having some trouble running rspec specs with js: true.

Version information:
qt: 4.8.6
rspec: 2.14.0
capybara: 2.2.0
capybara-webkit: 1.1.0
Ruby: 2.0.0p353
Rails: 3.2.16
OSX: Mavericks (OSX 10.9)

My specs seem to fail on `visit`. I receive the following:

```
Sign up form module
  Client side validation
    name field
      under 3 characters
        should have css "#user_name_input.error"
    email field
      invalid email address
        should have css "#user_email_input.error"
      unavailable email address

An error occurred in an after hook
  Errno::ECONNRESET: Connection reset by peer
  occurred at /Users/adamwaite/.rvm/gems/ruby-2.0.0-p353/gems/capybara-webkit-1.1.0/lib/capybara/webkit/connection.rb:37:in `gets'

        example at ./spec/features/partials/modules/users/sign_up_form_spec.rb:170 (FAILED - 1)
    username field
      under 5 characters

An error occurred in an after hook
  Errno::EPIPE: Broken pipe
  occurred at /Users/adamwaite/.rvm/gems/ruby-2.0.0-p353/gems/capybara-webkit-1.1.0/lib/capybara/webkit/connection.rb:29:in `write'

        example at ./spec/features/partials/modules/users/sign_up_form_spec.rb:180 (FAILED - 2)
      invalid characters

An error occurred in an after hook
  Errno::EPIPE: Broken pipe
  occurred at /Users/adamwaite/.rvm/gems/ruby-2.0.0-p353/gems/capybara-webkit-1.1.0/lib/capybara/webkit/connection.rb:29:in `write'

        example at ./spec/features/partials/modules/users/sign_up_form_spec.rb:187 (FAILED - 3)
```

All subsequent tests fail with `Errno::EPIPE: Broken pipe`.

What's odd is that if I run the failing specs again (a feature in the new spec to rerun the failures) then it all goes green.

```
Sign up form module
  Client side validation
    name field
      under 3 characters
        should have css "#user_name_input.error"
    email field
      invalid email address
        should have css "#user_email_input.error"
      unavailable email address
        should have css "#user_email_input.error"
    username field
      under 5 characters
        should have css "#user_username_input.error"
      invalid characters
        should have css "#user_username_input.error"
      unavailable username
        should have css "#user_username_input.error"
    password field
      under 6 characters
        should have css "#user_password_input.error"
      without a numerical character
        should have css "#user_password_input.error"
    password confirmation
      doesn't match the password
        should have css "#user_password_confirmation_input.error"
    postcode
      under 4 characters
        should have css "#user_postcode_input.error"

Finished in 13.81 seconds
10 examples, 0 failures
```

So the failing specs pass the second time around.

I have seen this sort of thing on some of capybara-webkit's previous issues and it was related to fonts and font-awesome from what I gathered. I attempted numerous fixes described by those issues but had no luck. The fact the specs pass on a second run perhaps suggest it isn't related to fonts though?

Any help or pointers where to look would be great.

**Comment by calleerlandsson** [comment]
I'll close this issue due to the long time of inactivity. Please re-open it if you're still seeing these issues using the latest version of capybara-webkit.

**Comment by ArronHoltzhausen** [comment]
this is still persisting 
Connection reset by peer - recvfrom(2) (Errno::ECONNRESET)
 Broken pipe (Errno::EPIPE)

**Comment by calleerlandsson** [comment]
Hey @ArronHoltzhausen! What versions of the following software are you running?

- capybara-webkit
- Qt
- Ruby
- Mac OS / Linux / BSD / Windows

**Comment by antmachine** [comment]
Similar issue here.

ruby 2.2.4
QT 5
Ubuntu 14.04.4 LTS
capybara-webkit 1.10.1

```ruby/2.2.0/gems/capybara-webkit-1.10.1/lib/capybara/webkit/connection.rb:22:in `write'

**Comment by gabelilly** [comment]
The issue I was having was caused by having multiple capybara-webkit Jenkins jobs running at the same time. I was using the headless gem, but not giving it any custom args. I believe what was happening is a long job would start up, then a shorter job would start and finish while the long job was still running. The short job would destroy the xvfb session, which would crash the long job. 

I saw many recommendations to move to xvfb-run. After messing with the settings, I was able to get jobs starting, running, and finishing at the same time without causing any issues. Here's the command I am using in the Jenkins jobs.

    xvfb-run -d --server-args="-screen 0 1024x768x24" bundle exec rspec

Hope this helps someone else.
